relay-dynamic-config/src/feature.rs

```diff
@@ -99,6 +99,13 @@ pub enum Feature {
     #[serde(rename = "organizations:indexed-spans-extraction")]
     ExtractSpansFromEvent,
 
+    /// Enables description scrubbing for MongoDB spans (and consequently, their presence in the
+    /// Queries module inside Sentry).
+    ///
+    /// Serialized as `organizations:performance-queries-mongodb-extraction`.
+    #[serde(rename = "organizations:performance-queries-mongodb-extraction")]
+    ScrubMongoDBDescriptions,
```

Comment:
Does this feature make sense the way it is right now? 

It seems like `mongodb` is unconditionally enabled with this PR only the scrubbing is conditional. For me this seems like we should have a feature flag to enable/disable mongodb alltogether not just the scrubbing which is supposed to help us with cardinality.

Reply:
Here's my understanding.

There are two places that Relay currently special-cases MongoDB. The first is disabled databases.

https://github.com/getsentry/relay/blob/70c15da848216cf5a20062393d47d1d10341323f/relay-dynamic-config/src/defaults.rs#L11-L17
https://github.com/getsentry/relay/blob/70c15da848216cf5a20062393d47d1d10341323f/relay-dynamic-config/src/defaults.rs#L123-L126

`is_db` is used to control the presence of certain metrics and tags in [`hardcoded_span_metrics`](https://github.com/getsentry/relay/blob/70c15da848216cf5a20062393d47d1d10341323f/relay-dynamic-config/src/defaults.rs#L120).

This PR removes that check, which means we will produce more metrics. However, note that:
- a number of SDKs (including v8 of the JS SDK) don't actually have `mongodb` as part of the span `op`, and were never subject to this check
- high cardinality of the resulting metrics comes from the `span.description` tag, which is protected against separately (see following)

The second existing special case is in the span description scrubbing.

https://github.com/getsentry/relay/blob/70c15da848216cf5a20062393d47d1d10341323f/relay-event-normalization/src/normalize/span/description/mod.rs#L63-L70

We check again for `mongodb` in the op (which as I mentioned, is often already bypassed), *but* the `is_sql_mongodb` function catches them all at this point by [checking for a `db.system` of `mongodb` or JSON-looking description](https://github.com/getsentry/relay/blob/70c15da848216cf5a20062393d47d1d10341323f/relay-event-normalization/src/normalize/span/description/mod.rs#L146-L152). This is the handling we fall through to when the feature is off.

---

So:
- This PR will change the situation from "some MongoDB spans are counted as `is_db`" to "_all_ MongoDB spans are counted as `is_db`", but
- There will be no increase in cardinality, as all MongoDB span descriptions will continue to be `None` in cases where the flag is not set.

In an ideal world, I would have kept the behaviour exactly the same and only conditionally changed `DISABLED_DATABASES`. But, looking at how `hardcoded_span_metrics`, I can't see any way to get feature flags down to it. So I reasoned that the outcomes above were probably acceptable, but I could definitely be wrong! Please let me know what you think. Thanks so much 🙏